baseline-error-prone/src/test/java/com/palantir/baseline/errorprone/StrictCollectionIncompatibleTypeTest.java

```diff
@@ -0,0 +1,169 @@
+                "   }",
+                "   String f4(CustomMap map, String key) {",
+                "       // BUG: Diagnostic contains: Value 'key' of type 'String' is not "
+                        + "compatible with the expected type 'Integer'",
+                "       return map.remove(key);",
+                "   }",
+                "   Object f4(Map<Integer, String> map, List<String> keys) {",
+                "       // BUG: Diagnostic contains: Type 'String' is not compatible with the expected type 'Integer'",
+                "       return keys.stream().map(map::get);",
+                "   }",
+                "   Object f6(Map<CharSequence, String> map, List<String> keys) {",
+                "       return keys.stream().map(map::get);",
+                "   }",
+                "   Object f7(Map<String, String> map, List<CharSequence> keys) {",
+                "       return keys.stream().map(map::get);",
+                "   }",
+                "   Converter f8(CustomMap map, List<String> keys) {",
+                "       // This should fail once we implement a more general check",
+                "       return map::get;",
+                "   }",
+                "   interface CustomMap extends Map<Integer, String> {}",
+                "   interface Converter { String convert(Integer input); }",
+                "}"
+        ).doTest();
+    }
+
+    @Test
+    void testUnexpectedType_collection() {
+        helper().addSourceLines(
+                "Test.java",
+                "import java.util.Collection;",
+                "import java.util.List;",
+                "import com.google.common.collect.ImmutableMap;",
+                "class Test {",
+                "   boolean f0(Collection<Integer> in, String key) {",
+                "       // BUG: Diagnostic contains: incompatible types",
+                "       return in.contains(key);",
+                "   }",
+                "   boolean f1(List<Integer> in, String key) {",
+                "       // BUG: Diagnostic contains: incompatible types",
+                "       return in.contains(key);",
+                "   }",
+                "   boolean f2(Custom in, Integer key) {",
+                "       // BUG: Diagnostic contains: incompatible types",
+                "       return in.contains(key);",
+                "   }",
+                "   boolean f3(ImmutableMap<Integer, String> in, String key) {",
+                "       // BUG: Diagnostic contains: incompatible types",
+                "       return in.keySet().contains(key);",
+                "   }",
+                "   boolean f4(List<Integer> in, String key) {",
+                "       // BUG: Diagnostic contains: incompatible types",
+                "       return in.remove(key);",
+                "   }",
+                "   boolean f4(List<Integer> in) {",
+                "       // BUG: Diagnostic contains: incompatible types",
+                "       return in.remove(5L);",
```

Comment:
This tests boxing the wrong primitive results in a failure

Reply:
oh somehow glossed over these, I was scanning for a function with a type of `int` or something